libraries/AP_HAL/system.h

```diff
@@ -28,4 +28,48 @@ uint64_t native_millis64();
 void dump_stack_trace();
 void dump_core_file();
 
+template <typename T>
+inline bool timeout_expired(const T past_time, const T now, const T timeout)
+{
+    const T dt = now - past_time;
+    return (dt > timeout);
+}
+
+template <typename T>
+inline T timeout_remaining(const T past_time, const T now, const T timeout)
+{
+    const T dt = now - past_time;
+    return (dt > timeout) ? T(0) : (timeout - dt);
+}
+
+inline bool timeout_expired_us(uint32_t past_time_us, uint32_t timeout_us)
+{
+    return timeout_expired(past_time_us, micros(), timeout_us);
+}
+
+inline uint32_t timeout_remaining_us(uint32_t past_time_us, uint32_t timeout_us)
+{
+    return timeout_remaining(past_time_us, micros(), timeout_us);
+}
+
+inline bool timeout_expired_us(uint64_t past_time_us, uint64_t timeout_us)
```

Comment:
timeout_remaining_us64()

Reply:
actually this stopped the overloading I wanted to allow so didn't do this